### PR TITLE
Cup game task

### PR DESCRIPTION
**Pull request by MaxRobinsonTheGreat**

### CODE REVIEW

File: docs/packages/docs/tasks/cup-game-task.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+The cup game task only works in the CupGame world in the dexterity package.
+The game will not start if a cup game task is not added to an agent.
+
+Configuration
+-------------
+
+Each of the following parameters can be placed in the configuration field
+for a cup game task sensor (see :ref:`scenario files <scenario-files>`.)
+The configuration can also be changed programmatically by using
```
Comment (jaydenmilne) [inline docs/packages/docs/tasks/cup-game-task.rst]: Sentence fragment

File: tests/sensors/test_cup_game_sensors.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                                                   # binary_path=binary_path,
+                                                   # show_viewport=False,
+                                                   start_world=False,
+                                                   uuid="") as env:
+
+        env.reset()
+
+        for _ in range(300):
+            _ = env.tick()
+        env.agents["sphere0"].teleport([-.4, -.9, 1.8], [0, 0, 90])
```
Comment (jaydenmilne) [inline tests/sensors/test_cup_game_sensors.py]: Fix rotation discrepancy and remove this line if needed

File: tests/sensors/test_cup_game_sensors.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                "location": [-.4, -.9, 1.8],
+                "rotation": [90, 0, 0]
+            }
+        ],
+        "window_width": 1024,
+        "window_height": 1024
+    }
+
+
+def test_cup_game_sensors():
```
Comment (jaydenmilne) [inline tests/sensors/test_cup_game_sensors.py]: Name suggestion: `test_ball_location_sensor_and_reward`

File: tests/sensors/test_ball_location_and_reward.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    with holodeck.environments.HolodeckEnvironment(scenario=cfg,
+                                                   # binary_path=binary_path,
+                                                   # show_viewport=False,
+                                                   start_world=False,
+                                                   uuid="") as env:
+
+        env.reset()
+
+        for _ in range(300):
```
Comment (jaydenmilne) [inline tests/sensors/test_ball_location_and_reward.py]: `env.tick(300)`

File: tests/sensors/test_ball_location_and_reward.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            _ = env.tick()
+        state = None
+        touched_cup = False
```
Comment (jaydenmilne) [inline tests/sensors/test_ball_location_and_reward.py]: FYI you don't have to do this, this isn't how Python's scoping rules work. This, for example, is fine:

```python
def tacoify():
	for x in range(10):
		burrito = x

	print(x)
	print(burrito)
	
tacoify()
```

See https://python-textbok.readthedocs.io/en/1.0/Variables_and_Scope.html#variable-scope-and-lifetime

File: tests/sensors/test_ball_location_and_reward.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        for _ in range(30):
+            state, reward, terminal, _ = env.step([0])
+            if reward == 1:
+                touched_cup = True
+        assert touched_cup and state["BallLocationSensor"] == 2
+        env.tick(1000)
```
Comment (jaydenmilne) [inline tests/sensors/test_ball_location_and_reward.py]: Delete this?

File: tests/sensors/test_ball_location_and_reward.py (hunk 5 above)
Comment (MaxRobinsonTheGreat) [reply tests/sensors/test_ball_location_and_reward.py]: pycharm gets mad at me when I do that but you're right

**Pull request merged**